File: servers/japronto-graphene/schema.py
from graphene import ID, Interface, ObjectType, Schema, List, Field, String

class Base(Interface):
    id = ID()
    def resolve_id(self, info):
        return "id"


class Query(ObjectType):
    class Meta:
        interfaces = (Base, )

    string = String()
    list_of_strings = List(String)
    list_of_objects = List(lambda: Query)
    list_of_interfaces = List(Base)

    def resolve_string(self, info):
        return string

    def resolve_list_of_strings(self, info):
        return list_of_strings

    def resolve_list_of_objects(self, info):
        return list_of_objects

    def resolve_list_of_interfaces(self, info):
        return list_of_objects


string = "Hello World!"
list_of_strings = (string, ) * 100
list_of_objects = (Query(), ) * 100


schema = Schema(query=Query)

File: servers/node-express/schema.js
var {
  GraphQLSchema,
  GraphQLObjectType,
  GraphQLString,
  GraphQLInterfaceType,
  GraphQLList
} = require("graphql");

var string = "Hello World!";
var listOfStrings = Array(100).fill(string);
var listOfObjects = Array(100).fill({});

var Base = new GraphQLInterfaceType({
  name: "Base",
  fields: {
    id: {
      type: GraphQLString
    }
  }
});
var Query = new GraphQLObjectType({
  name: "Query",
  interfaces: [Base],
  fields: () => ({
    id: {
      type: GraphQLString,
      resolve: () => "id"
    },
    string: {
      type: GraphQLString,
      resolve: () => string
    },
    listOfStrings: {
      type: GraphQLList(GraphQLString),
      resolve: () => listOfStrings
    },
    listOfObjects: {
      type: GraphQLList(Query),
      resolve: () => listOfObjects
    },
    listOfInterfaces: {
      type: Base,
      resolve: () => listOfObjects
    }
  })
});

schema = new GraphQLSchema({ query: Query });
module.exports = schema;
